server/src/resolvers/employeeSkill.ts

```diff
@@ -34,23 +33,28 @@ export class EmployeeSkillResolver {
       };
     }
 
-    const skill = await Skills.findOne({ skillName: input.skillName });
-    if (!skill) {
+    const experienceLevels = ["Beginner", "Intermediate", "Expert"];
+    if (!experienceLevels.includes(input.experienceLevel)) {
       return {
         errors: [
           {
-            field: "skillName",
-            message: "Can not find this skill",
+            field: "experienceLevel",
+            message: "Invalid Input",
           },
         ],
       };
     }
+    var skill = await Skills.findOne({ skillName: input.skillName });
+    if (!skill) {
+      skill = await Skills.create({ skillName: input.skillName }).save();
```

Comment:
forward the creation to the `SkillResolver`, don't create it manually here. (single source of truth)